### PR TITLE
feat: add prefix when using config_etcd:getkeys

### PR DESCRIPTION
**Pull request by billnote**
### What this PR does / why we need it:

fix: https://github.com/apache/apisix/issues/4552

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/prometheus/exporter.lua [hunk 1]
```diff
@@ -308,7 +308,13 @@ function _M.collect()
                            "processing metrics endpoint: ", err)
         end
 
-        local res, _ = config:getkey("/routes")
+        local key = "/routes/"
+        local local_conf = core.config.local_conf()
```
Comment (spacewander) [inline apisix/plugins/prometheus/exporter.lua:312]: Better to move it into getkey

**Comment by spacewander** [comment]
Where do you find "etcd key not found"  in the error log?

After consulting with the author of this feature, look like it doesn't require a real key. Actually, request any key from etcd will return the "X-Etcd-Index". A non-existed key is preferred because it doesn't return too much data, as we only care about the header.

**Comment by billnote** [comment]
Sorry, I check the code, This waring will only appear in our internal version.

But  getKey function  don't need add prefix automatically?  I will close this PR, if we don't need.

File: apisix/core/config_etcd.lua [hunk 2]
```diff
@@ -71,6 +71,11 @@ local function getkey(etcd_cli, key)
         return nil, "not inited"
     end
 
+    local local_conf, _ = config_local.local_conf()
```
Comment (spacewander) [inline apisix/core/config_etcd.lua:74]:
```suggestion
    local local_conf = config_local.local_conf()
```

File: apisix/core/config_etcd.lua [hunk 3]
```diff
@@ -71,6 +71,11 @@ local function getkey(etcd_cli, key)
...
+    if local_conf and local_conf.etcd and local_conf.etcd.prefix then
+        key = local_conf.etcd.prefix .. key
+    end
```
Comment (spacewander) [inline apisix/core/config_etcd.lua:77]: Should be added to _M.getkey?

File: apisix/plugins/prometheus/exporter.lua [hunk 4]
```diff
@@ -308,7 +308,7 @@ function _M.collect()
                            "processing metrics endpoint: ", err)
         end
 
-        local res, _ = config:getkey("/routes")
+        local res, _ = config:getkey("/routes/")
```
Comment (spacewander) [inline apisix/plugins/prometheus/exporter.lua:311]: Let's change the key name.

File: apisix/core/config_etcd.lua (hunk 3 above)
Comment (billnote) [reply apisix/core/config_etcd.lua:77]: ok

**Review by spacewander — COMMENTED** [review]
Need a test in https://github.com/apache/apisix/blob/master/t/core/config_etcd.t

File: apisix/plugins/prometheus/exporter.lua [hunk 5]
```diff
@@ -308,7 +308,8 @@ function _M.collect()
                            "processing metrics endpoint: ", err)
         end
 
-        local res, _ = config:getkey("/routes")
+        -- use phantom key to get etcd index
```
Comment (spacewander) [inline apisix/plugins/prometheus/exporter.lua:311]: The comment is too short compared with my explanation

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**